documentation/docs/14-content-security-policy.md

```diff
@@ -0,0 +1,53 @@
+---
+title: Content Security Policy
+---
+
+At the moment, SvelteKit supports adding Content Security Policy via hooks. In environments with a runtime, HTTP headers can be added to the response object.
+
+However, SvelteKit also requires some small pieces of inline JavaScript in order for hydration to work. To avoid using `'unsafe-inline'` (which, as the name suggests, should be avoided), SvelteKit can be configured to inject CSP nonces into the HTML it generates.
+
+The nonce value is availiable to hooks as `request.locals.nonce`. A basic CSP handler hook might then look like this:
+
+```javascript
+export async function handle ({ request, resolve }) {
+  const directives = {
+    'default-src': ["'self'", 'static.someotherdomain.com'],
+    'script-src': ["'strict-dynamic'"],
+    'style-src': ["'self'"]
+  };
+  const response = await resolve(request);
+
+  if (response.headers['content-type'] !== 'text/html') {
+  	return response
+	}
+
+  const nonce = request.locals.nonce;
+
+  directives['script-src'].push(`'nonce-${nonce}'`);
+  directives['style-src'].push(`'nonce-${nonce}'`);
+
+  if (process.env.NODE_ENV === 'development') {
```

Comment:
Is this needed because of https://github.com/sveltejs/svelte/issues/6662? If so, it might be nice to add a comment here with a link to the issue. That issue doesn't mention anything about it being only for dev mode, so I'm wondering if that should be added to the issue description or we should remove this `if` check?

Reply:
Oh, I see the explanation down below on line 47. I'd move it up here and make it a comment